file/fs.lisp

```diff
@@ -655,6 +657,13 @@ NAMESTRING as the second."
     (apply #'delete-file-using-host (pathname-host path) path args))
   t)
 
+(defgeneric delete-directory-using-host (host path recursive))
```

Comment:
This should take &key like delete-file. Allows for file-system-specific behaviour.
delete-directory should forward all arguments on with apply.
(defgeneric delete-directory-using-host (host path &key recursive)) will require methods to accept :recursive